File: 1359-count-all-valid-pickup-and-delivery-options/1359-count-all-valid-pickup-and-delivery-options.cs
public class Solution {
    private const int MOD = 1000000007;

    public int CountOrders(int n) {
        long count = 1;
        for (int i = 2; i <= n; i++) {
            count = (count * (2 * i - 1) * i) % MOD;
        }
        return (int)count;
    }

}

File: 1359-count-all-valid-pickup-and-delivery-options/1359-count-all-valid-pickup-and-delivery-options.py
class Solution:
    def countOrders(self, n: int) -> int:
        dp = [0]*n
        dp[0] = 1
     
        
        for i in range(2,n+1):
            n = 2*(i - 1) + 1
            curr_count = (n * (n + 1))// 2
           
            dp[i - 1] += (dp[i - 2] * curr_count)
            
        
        return dp[-1] % (10**9 + 7)
            
            